### PR TITLE
Refactoring PostgreRepository

### LINKED ISSUE #589: Repository module refactoring
## Description:
Due to the heavy support for the module, it needs to be refactored.
Move common elements to separate files, rename objects. Split it into subfolders.

## Acceptance Criteria:
The workability has not changed

### PR DESCRIPTION
**Pull request by memorisecodead**
Next PostgreRepositoryManager cutting

### CODE REVIEW

**Review by slavexx — APPROVED** [review]

File: DataAccess/DataAccess.Postgre/PostgreQuery.hpp [hunk 1]
```diff
@@ -6,24 +6,24 @@
...
 /**
- * @class PostgreTable
- * @brief Postgre table class.
+ * @class PostgreQuery
+ * @brief Postgre query class.
  * @details You can see some examples below for how to use it.
  * @code
- *    PostgreTable("tableName1").Select()->columns({"column1", "column2",
+ *    PostgreQuery("tableName1").Select()->columns({"column1", "column2",
  * ...})->where("condition")->...->execute()/OR/getQuery(); PostgreTable("tableName2").Insert()->field(1,
```
Comment (ivanboika) [inline DataAccess/DataAccess.Postgre/PostgreQuery.hpp]: You forgot to change there

Comment (memorisecodead) [reply DataAccess/DataAccess.Postgre/PostgreQuery.hpp]: See, thank you for review!

**Review by ivanboika — APPROVED** [review]

File: DataAccess/Public/Include/DataAccess/IRepository.hpp [hunk 2]
```diff
@@ -14,84 +15,94 @@ namespace DataAccess
...
 {
 public:
+
     /**
-     * @brief Draft method for getting all channel list from the server.
-     * @return channels list as vector of ChannelInfo.
-     */
+    * @brief draft method for getting all channel list from the server.
+    * @return channels list as vector of ChannelInfo.
+    */
```
Comment (slavexx) [inline DataAccess/Public/Include/DataAccess/IRepository.hpp]: Why did you change letter? This is the beginning of sentence in doxygen documentation.
image

File: DataAccess/Public/Include/DataAccess/IServerRepositories.hpp [hunk 3]
```diff
@@ -10,26 +11,56 @@
...
 /**
- * @brief The IMessagesRepository abstract subclass from IMasterRepository
+ * @brief The IMessagesRepository abstract subclass from IMasterRepository.
  */
 struct IMessagesRepository : IMasterRepository
 {
-    /// Pure Virtual method to get Message History
+    /**
+    * @brief virtual method for getting history. \
+    * @param channelID - channel's id from which we want to get message history.
```
Comment (slavexx) [inline DataAccess/Public/Include/DataAccess/IServerRepositories.hpp]: I think, in 19th line should be '/'

**Review by slavexx — APPROVED** [review]
A couple of not very significant moments

File: DataAccess/Public/Include/DataAccess/IServerRepositories.hpp (hunk 3 above)
Comment (memorisecodead) [reply DataAccess/Public/Include/DataAccess/IServerRepositories.hpp]: I haven't touched that point, but I can fix it.

File: DataAccess/Public/Include/DataAccess/IServerRepositories.hpp [hunk 4]
```diff
@@ -53,22 +94,45 @@ struct IRepliesRepository : IMasterRepository
...
+    /**
+    * @brief  method for deleting channel.
+    * @return special code for success or failed deleting;
+    */
     virtual Utility::ChannelDeleteCode deleteChannel(const Models::ChannelDeleteInfo& channel) = 0;
 
-    /// Virtual method to create channel
+    /**
+    * @brief  method for creating channel.
+    * @return special code for success or failed creating;
```
Comment (ivanboika) [inline DataAccess/Public/Include/DataAccess/IServerRepositories.hpp]: That's a code smell which means that using at the end of comment line ';' is a bad idea because it's not that doxygen can detect as documentable symbol

**Review by ivanboika — APPROVED** [review]
Push it in dev
